File: app/src/main/java/com/appdate/raissequeiroz/fragment/DatePickerDialogFragment.java
package com.appdate.raissequeiroz.fragment;

import android.app.Activity;
import android.app.DatePickerDialog;
import android.app.Dialog;
import android.os.Bundle;
import android.widget.DatePicker;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;

import com.appdate.raissequeiroz.AddActivity;
import com.appdate.raissequeiroz.R;
import com.appdate.raissequeiroz.util.DateUtils;
import com.google.android.material.textfield.TextInputEditText;

import java.util.Calendar;

public class DatePickerDialogFragment extends DialogFragment implements
        DatePickerDialog.OnDateSetListener {

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        Calendar calendar = Calendar.getInstance();
        int ano =  calendar.get(Calendar.YEAR);
        int mes =  calendar.get(Calendar.MONTH);
        int dia =  calendar.get(Calendar.DAY_OF_MONTH);
        return new DatePickerDialog(getActivity(), this, ano, mes, dia);
    }

    @Override
    public void onDateSet(DatePicker view, int year, int month, int dayOfMonth) {
        view.setMinDate(Calendar.getInstance().getTimeInMillis());

        TextInputEditText inputData = getActivity().findViewById(R.id.inputData);
        inputData.setText(DateUtils.dateToString(year, month, dayOfMonth));

        inputData.setError(null);
    }

}

File: app/src/main/java/com/appdate/agendamento/fragment/DatePickerDialogFragment.kt
package com.appdate.agendamento.fragment

import android.app.DatePickerDialog
import android.app.DatePickerDialog.OnDateSetListener
import android.app.Dialog
import android.os.Bundle
import android.widget.DatePicker
import androidx.fragment.app.DialogFragment
import com.appdate.agendamento.R
import com.appdate.agendamento.util.DateUtils
import com.google.android.material.textfield.TextInputEditText
import java.util.*

class DatePickerDialogFragment : DialogFragment(), OnDateSetListener {
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val calendar = Calendar.getInstance()
        val ano = calendar[Calendar.YEAR]
        val mes = calendar[Calendar.MONTH]
        val dia = calendar[Calendar.DAY_OF_MONTH]
        return DatePickerDialog(activity!!, this, ano, mes, dia)
    }

    override fun onDateSet(view: DatePicker, year: Int, month: Int, dayOfMonth: Int) {
        val inputData: TextInputEditText = activity!!.findViewById(R.id.inputData)
        inputData.setText(DateUtils.dateToString(year, month, dayOfMonth))
        inputData.error = null
    }
}